src/javascript/javascript-import-scanner.ts

```diff
@@ -36,44 +50,112 @@ export class JavaScriptImportScanner implements JavaScriptScanner {
         if (node.callee.type as string !== 'Import') {
           return;
         }
-        const arg = node.arguments[0]! as any;
-        if (!arg) {
-          // TODO(usergenic): push a warning
+        if (node.arguments.length !== 1) {
+          warnings.push(new Warning({
+            message: 'Malformed import',
+            sourceRange: document.sourceRangeForNode(node)!,
+            severity: Severity.WARNING,
+            code: 'malformed-import',
+            parsedDocument: document,
+          }));
           return;
         }
-        const source = arg.value as FileRelativeUrl;
-        if (!isPathImport(source)) {
-          // TODO(usergenic): push a warning
+        const arg = node.arguments[0];
+        if (arg.type !== 'StringLiteral') {
+          warnings.push(new Warning({
+            message:
+                'Cannot analyze dynamic imports with non-literal arguments',
+            sourceRange: document.sourceRangeForNode(node)!,
+            severity: Severity.WARNING,
+            code: 'non-literal-import',
+            parsedDocument: document,
+          }));
           return;
         }
         imports.push(new ScannedImport(
             'js-import',
-            source,
+            scanner._resolveSpecifier(
```

Comment:
Why not emit a warning and return if we can't resolve the specifier?

Do we have a use for a ScannedImport without a url?

Reply:
The import still exists, we just couldn't determine a valid URL for it. It should be useful for linting. We don't produce an `Import` if `url` is undefined, which is consistent with current behavior, but I feel a little weird about that too.